anyway/infographics_utils.py

```diff
@@ -978,29 +1019,352 @@ def top_road_segments_accidents_mock_data():  # Temporary for Frontend
+    elif widget_id == WidgetId.injured_count_by_accident_year:
+        res = InjuredCountByAccidentYearWidget(request_params)
+    elif widget_id == WidgetId.accident_count_by_day_night:
+        res = AccidentCountByDayNightWidget(request_params)
+    elif widget_id == WidgetId.accident_count_by_hour:
+        res = AccidentCountByHourWidget(request_params)
+    elif widget_id == WidgetId.accident_count_by_road_light:
+        res = AccidentCountByRoadLightWidget(request_params)
+    elif widget_id == WidgetId.top_road_segments_accidents_per_km:
+        res = TopRoadSegmentsAccidentsPerKmWidget(request_params)
+    elif widget_id == WidgetId.injured_count_per_age_group:
+        res = InjuredCountPerAgeGroupWidget(request_params)
+    elif widget_id == WidgetId.vision_zero:
+        res = VisionZeroWidget(request_params)
+    elif widget_id == WidgetId.accident_count_by_driver_type:
+        res = AccidentCountByDriverTypeWidget(request_params)
+    elif widget_id == WidgetId.accident_count_by_car_type:
+        res = AccidentCountByCarTypeWidget(request_params)
+    elif widget_id == WidgetId.injured_accidents_with_pedestrians:
+        res = InjuredAccidentsWithPedestriansWidget(request_params)
+    elif widget_id == WidgetId.accident_severity_by_cross_location:
+        res = AccidentSeverityByCrossLocationWidget(request_params)
+    elif widget_id == WidgetId.motorcycle_accidents_vs_all_accidents:
+        res = MotorcycleAccidentsVsAllAccidentsWidget(request_params)
+    elif widget_id == WidgetId.accident_count_pedestrians_per_vehicle_street_vs_all:
+        res = AccidentCountPedestriansPerVehicleStreetVsAllWidget(request_params)
+    elif widget_id == WidgetId.top_road_segments_accidents:
+        res = TopRoadSegmentsAccidentsWidget(request_params)
+    elif widget_id == WidgetId.pedestrian_injured_in_junctions:
+        res = PedestrianInjuredInJunctionsWidget(request_params)
+    else:
+        logging.error(f"create_widget: failed to create:{widget_id}")
+        res = None
+    return res
+
+
+def extract_news_flash_location(news_flash_obj):
+    resolution = news_flash_obj.resolution if news_flash_obj.resolution else None
+    if not news_flash_obj or not resolution or resolution not in resolution_dict:
+        logging.warning(f"could not find valid resolution for news flash id {str(news_flash_obj.id)}")
+        return None
+    data = {"resolution": resolution}
+    for field in resolution_dict[resolution]:
+        curr_field = getattr(news_flash_obj, field)
+        if curr_field is not None:
+            data[field] = curr_field
+    gps = {}
+    for field in ["lon", "lat"]:
+        gps[field] = getattr(news_flash_obj, field)
+    return {"name": "location", "data": data, "gps": gps}
+
+
+def get_query(table_obj, filters, start_time, end_time):
+    query = db.session.query(table_obj)
+    if start_time:
+        query = query.filter(getattr(table_obj, "accident_timestamp") >= start_time)
```

Comment:
I don't understand the need for `getattr()` call here.

Reply:
Me neither... (I did not wrote this code, just packed it into classes.